lib/elixir/lib/file/stream.ex

```diff
@@ -93,27 +104,20 @@ defmodule File.Stream do
       Stream.resource(start_fun, next_fun, &:file.close/1).(acc, fun)
     end
 
-    def count(%{path: path, modes: modes, line_or_bytes: :line} = stream) do
+    def count(%{modes: modes, line_or_bytes: :line, path: path} = stream) do
       pattern = :binary.compile_pattern("\n")
       counter = &count_lines(&1, path, pattern, read_function(stream), 0)
-
-      case File.open(path, read_modes(modes), counter) do
-        {:ok, count} ->
-          {:ok, count}
-
-        {:error, reason} ->
-          raise File.Error, reason: reason, action: "stream", path: path
-      end
+      {:ok, open!(stream, modes, counter)}
     end
 
-    def count(%{path: path, line_or_bytes: bytes, raw: true, modes: modes}) do
-      case File.stat(path) do
+    def count(%{path: path, line_or_bytes: bytes, raw: true, modes: modes, node: node} = stream) do
+      case :erpc.call(node, File, :stat, [path]) do
```

Comment:
(Sorry might be a naive question) Is this OK to replace this by `:erpc.call` even in the case the node is the current one? I'm not sure if there is any downside/overhead and if it is worth optimizing or not.

Reply:
:erpc is a no-op if the node is the current one. :)

Reply:
Thanks for confirming 👍
It seems to be [optimized](https://github.com/erlang/otp/blob/2c864f2f9ef61750006afde6af33453a11d025cd/lib/kernel/src/erpc.erl#L121) indeed but still seemed more involved than simply invoking the function, which is why I asked.